src/components/loginComponent.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'proptypes';
+import Input from './Input';
+import Loader from './Loader';
+import logo from '../assets/img/logo.png';
+
+class loginComponent extends Component {
+    state = {
+        email: '',
+        password: '',
+    }
+    
+    handleChange = (event) => {
+        this.setState({ [event.target.name]: event.target.value });
+    }
+
+    onLogin = (event) => {
+        event.preventDefault();
+        const { login } = this.props;
+        login(this.state)
+    }
+
+    render() {
+        const { loading, success, error, data } = this.props;
+        const { email, password } = this.state;
+
+        if (success) {
+            localStorage.setItem('user', JSON.stringify(data));
```

Comment:
It is not advisable to handle side effects inside components like this. Redux should handle the side effects. I would suggest have a redux middleware the sets the user in the local storage once the action succeeds. 
https://redux.js.org/advanced/middleware
```
const authMiddleware = () => next => (action) => {
  if (!action) return;
  if (action.type === `${types.LOGIN}_SUCCESS`) {
    const payloadData = action.payload.data;
    const { token } = payloadData;
    action.payload.token = token;
    localStorage.setItem('token', token);
  }
  if (action.type === types.LOGOUT) {
    localStorage.removeItem('token');
  }
  return next(action);
};
```

Reply:
On it.